go/chat/unfurl_test.go

```diff
@@ -194,31 +194,45 @@ func TestChatSrvUnfurl(t *testing.T) {
 			require.Error(t, err)
 			require.IsType(t, libkb.NotFoundError{}, err)
 			select {
-			case mu := <-listener0.messagesUnfurled:
-				require.Equal(t, 1, len(mu.Updates))
-				require.Equal(t, conv.Id, mu.Updates[0].ConvID)
-				require.Equal(t, origID, mu.Updates[0].Msg.GetMessageID())
-				require.True(t, mu.Updates[0].Msg.IsValid())
-				require.Equal(t, 1, len(mu.Updates[0].Msg.Valid().Unfurls))
-				typ, err := mu.Updates[0].Msg.Valid().Unfurls[0].UnfurlType()
-				require.NoError(t, err)
-				require.Equal(t, chat1.UnfurlType_GENERIC, typ)
-				generic := mu.Updates[0].Msg.Valid().Unfurls[0].Generic()
-				require.Nil(t, generic.Image)
-				require.NotNil(t, generic.Favicon)
-				require.NotZero(t, len(generic.Favicon.Url))
-				resp, err := http.Get(generic.Favicon.Url)
-				require.NoError(t, err)
-				defer resp.Body.Close()
-				var buf bytes.Buffer
-				_, err = io.Copy(&buf, resp.Body)
-				require.NoError(t, err)
-				refBytes, err := ioutil.ReadFile(filepath.Join("unfurl", "testcases", "nytimes_sol.ico"))
-				require.NoError(t, err)
-				require.True(t, bytes.Equal(refBytes, buf.Bytes()))
-				require.Equal(t, "MIKE", generic.Title)
-			case <-time.After(timeout):
-				require.Fail(t, "no message unfurl")
+			case <-listener0.newMessageRemote:
+				require.Fail(t, "no more messages")
+			default:
+			}
+			// We get two of these, one for local and remote, but its hard to know where they
```

Comment:
why cant' this use the local/remote differentiation the other updates have?

Reply:
I have a comment in there, it was just a lot of extra plumbing, this was easier.

Reply:
kk if there are future tests that need this double loop then it's probably worth it